core/class-extension-core.php

```diff
@@ -96,28 +96,58 @@ function install_notice() {
 			$plugin_installer = new PluginInstaller();
 			$screen           = get_current_screen();
 
-			// First lets check if WSAL is installed, but not active.
-			if ( $plugin_installer->is_plugin_installed( 'wp-security-audit-log/wp-security-audit-log.php' ) && ! is_plugin_active( 'wp-security-audit-log/wp-security-audit-log.php' ) ) : ?>
-                <div class="notice notice-success is-dismissible wsal-installer-notice">
-					<?php
-					printf(
-						'<p>%1$s &nbsp;&nbsp;<button class="activate-addon button button-primary" data-plugin-slug="wp-security-audit-log/wp-security-audit-log.php" data-plugin-download-url="%2$s" data-plugins-network="%4$s" data-nonce="%3$s">%5$s</button><span class="spinner" style="display: none; visibility: visible; float: none; margin: 0 0 0 8px;"></span></p>',
-						esc_html__( 'WP Activity Log is installed but not active.', 'wsal-extension-core' ),
-						esc_url( 'https://downloads.wordpress.org/plugin/wp-security-audit-log.latest-stable.zip' ),
-						esc_attr( wp_create_nonce( 'wsal-install-addon' ) ),
-						( is_a( $screen, '\WP_Screen' ) && isset( $screen->id ) && 'plugins-network' === $screen->id ) ? true : false, // confirms if we are on a network or not.
-						esc_html__( 'Activate WP Activity Log.', 'wp-security-audit-log' )
-					);
+			$freeInstalled = false;
+			$premiumInstalled = false;
+			$freeActivated = false;
+			$premiumActivated = false;
+
+			/* Starting checks */
+			/* Is there free version installed */
+			if ( $plugin_installer->is_plugin_installed( 'wp-security-audit-log/wp-security-audit-log.php' )) {
+				$freeInstalled = true;
+			}
+			/* Is there premium version installed */
+			if ( $plugin_installer->is_plugin_installed( 'wp-security-audit-log-premium/wp-security-audit-log.php' )) {
+				$premiumInstalled = true;
+			}
+			/* Is free version activated */
+			if (is_plugin_active( 'wp-security-audit-log/wp-security-audit-log.php' )) {
+				$freeActivated = true;
+			}
+			/* Is premium version activated */
+			if (is_plugin_active( 'wp-security-audit-log-premium/wp-security-audit-log.php' )) {
+				$freeActivated = true;
+			}
+			/* End checks */
+
+			if ( $freeInstalled || $premiumInstalled ) {
+				/* We have plugin installed */
+				if ( $freeActivated || $premiumActivated ) {
+					/* There is installed and activated plugin - bounce */
+					exit();
```

Comment:
Why `exit()`? I think `return;` would be more suitable to let the rest of the request carry on.